app/src/main/java/org/simple/clinic/bloodsugar/unitselection/BloodSugarUnitSelectionDialog.kt

```diff
@@ -23,17 +40,87 @@ class BloodSugarUnitSelectionDialog : AppCompatDialogFragment(), BloodSugarUnitS
       fragment.show(fragmentManager, FRAGMENT_TAG)
     }
   }
-  
-  @SuppressLint("InflateParams")
+
+  private val screenDestroys = PublishSubject.create<ScreenDestroyed>()
+  private val dialogEvents = PublishSubject.create<BloodSugarUnitSelectionEvent>()
+  private val events by unsafeLazy {
+    val doneButton = (dialog as AlertDialog).getButton(DialogInterface.BUTTON_POSITIVE)
+    doneClicks(doneButton)
+  }
+
+  private fun doneClicks(doneButton: Button): Observable<BloodSugarUnitSelectionEvent> {
+
+    val radioIdToBloodSugarUnits = mapOf(
+        R.id.bloodSugarUnitMg to BloodSugarUnitPreference.Mg,
+        R.id.bloodSugarUnitMmol to BloodSugarUnitPreference.Mmol
+    )
+
+    return doneButton
+        .clicks()
+        .map {
+          val bloodSugarUnitSelectionValue = radioIdToBloodSugarUnits.getValue(bloodSugarUnitGroup.checkedRadioButtonId)
+          DoneClicked(bloodSugarUnitSelection = bloodSugarUnitSelectionValue)
+        }
+  }
+
+  private val delegate: MobiusDelegate<BloodSugarUnitSelectionModel, BloodSugarUnitSelectionEvent, BloodSugarUnitSelectionEffect> by unsafeLazy {
+
+    MobiusDelegate.forActivity(
+        events = dialogEvents.ofType(),
+        defaultModel = BloodSugarUnitSelectionModel(),
+        update = BloodSugarUnitSelectionUpdate(),
+        effectHandler = effectHandlerFactory.create(this).build()
+    )
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    delegate.onRestoreInstanceState(savedInstanceState)
+  }
+
+  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+    return layout
```

Comment:
Is this necessary? By default, it should return the same layout which the dialog already has, right?